### PR TITLE
BAT implementation

### PR DESCRIPTION
**Pull request by dlf57**
This should be the final inclusion of BAT. I also added CM and BOB to use utils/calcs and made everything float16 as I found no difference in initial performance comparisons between float16 and float64 for the reps.

### CODE REVIEW

**Comment by shivupa** [comment]
Tests will help in a future commit but this looks good!

File: chemreps/bat.py [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+        torsion_bag = {}
+
+        # grab bonds/nonbonds
+        for i in range(current_molecule.n_atom):
+            for j in range(i, current_molecule.n_atom):
+                atomi = current_molecule.sym[i]
+                atomj = current_molecule.sym[j]
+                zi = current_molecule.at_num[i]
+                zj = current_molecule.at_num[j]
+                if i == j:
```
[the remark this answers is not in the document]
Comment (dlf57) [reply chemreps/bat.py]: Thank you for catching this.  I had left this out early on but I think we need to include the nuclear charge bags as I believe in the paper they do.

[the remark this answers is not in the document]
Comment (dlf57) [reply chemreps/bat.py]: This should fix the issue with leaving out the atom bag.

[the remark this answers is not in the document]
Comment (amandadumi) [reply chemreps/bat.py]: I agree, this looks great!

**Pull request merged**